packages/core/src/browser/tree/tree-widget.ts

```diff
@@ -60,16 +62,25 @@ export const defaultTreeProps: TreeProps = {
 @injectable()
 export class TreeWidget extends VirtualWidget implements StatefulWidget {
 
+    protected decorations: Map<string, DecorationData[]> = new Map();
+
     constructor(
         @inject(TreeProps) readonly props: TreeProps,
         @inject(ITreeModel) readonly model: ITreeModel,
-        @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer
+        @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer,
+        @inject(TreeDecoratorService) protected readonly decoratorService: TreeDecoratorService
```

Comment:
please use property injection and postConstruct annotation to avoid breaking subclasses

Reply:
👍 I did that, but unfortunately, I had problems with the `postConstruct` in the `containers` reusing the base `treeContainer`. Anyway, I will do another iteration and try to figure out what caused the defect because your recommendation, the property injection is much better.

Reply:
What kind of? I meant a new method in this class:
```ts
@postConstruct()
protected init() {
   // start listening changes here
}
```

Reply:
I did exactly that, it worked fine until I moved some bindings to the `navigator` and others to the `markers` extension. Then I had an error from `inversify`. I will try once more, perhaps I just overlooked something during my refactoring.